include/SugarFolders/SugarFolders.php

```diff
@@ -561,17 +561,17 @@ public function retrieveFoldersForProcessing($user, $subscribed = true)
 
         $found = array();
         while ($a = $this->db->fetchByAssoc($r)) {
-            if ((($a['folder_type'] == $myEmailTypeString) ||
-                    ($a['folder_type'] == $myDraftsTypeString) ||
-                    ($a['folder_type'] == $mySentEmailTypeString)) &&
-                ($a['created_by'] != $current_user->id)
-            ) {
+            if($a['folder_type'] == $myEmailTypeString) {
```

Comment:
Query result filtered in a PHP loop which filtered by a sub-query in a sub-PHP-loop, the query more optimizable for e.g the $myEmailTypeString variable value can be only "inbound" which means it's a constant "WHERE folder_type = 'inbound'" somewhere at the end of the query. Below the $children is a sub-query which also can be an sql JOIN.